### PR TITLE
chore: Use `errgroup` over `goroutine.Parallel`

### PR DESCRIPTION
**Pull request by efritz**
Update use of custom Parallel function to use the x/sync/errgroup packaged used in other places.

## Test plan

Unit and integration tests have *incredible* coverage here!

### CODE REVIEW

File: enterprise/internal/codeintel/stores/dbstore/commits.go [hunk 1]
```diff
@@ -347,9 +347,12 @@ func (s *Store) calculateVisibleUploadsInternal(
 	graph := commitgraph.NewGraph(commitGraph, commitGraphView)
 
 	// Write the graph into temporary tables in Postgres
+	ctx, cancel := context.WithCancel(ctx)
 	if err := tx.writeVisibleUploads(ctx, sanitizeCommitInput(ctx, graph, refDescriptions, maxAgeForNonStaleBranches, maxAgeForNonStaleTags)); err != nil {
+		cancel() // wtf
```
Comment (LawnGnome) [inline enterprise/internal/codeintel/stores/dbstore/commits.go:352]: Could you elaborate WTF is up with the WTF?

File: enterprise/internal/codeintel/stores/dbstore/commits.go [hunk 2]
```diff
@@ -347,9 +347,12 @@ func (s *Store) calculateVisibleUploadsInternal(
...
 		return err
 	}
+	cancel() // wtf
```
Comment (LawnGnome) [inline enterprise/internal/codeintel/stores/dbstore/commits.go:355]: I mean, if we always have to cancel that _would_ be WTF.

**Review by LawnGnome — COMMENTED** [review]
A bit confused by the cancellations, but the actual `errgroup` change looks legit.

File: enterprise/internal/codeintel/stores/dbstore/commits.go (hunk 1 above)
Comment (efritz) [reply enterprise/internal/codeintel/stores/dbstore/commits.go:352]: DID NOT MEAN TO COMMIT

File: enterprise/internal/codeintel/stores/dbstore/commits.go (hunk 2 above)
Comment (efritz) [reply enterprise/internal/codeintel/stores/dbstore/commits.go:355]: DID NOT MEAN TO COMMIT

**Comment by efritz** [comment]
@LawnGnome Removed extra code I accidentally committed containing additional cancellations. That was just some local testing state I was playing around with.

**Review by Strum355 — APPROVED** [review]
:letsgo:

**Pull request merged**